server/controllers/users/changePassword.js

```diff
@@ -1,3 +1,22 @@
-exports.changePassword = (req, res) => {
-  res.send('hello from changePassword file');
+const { hash, compare } = require('bcrypt');
+const { users: { changePassword, getUserById } } = require('../../database/queries');
+
+exports.changePassword = (req, res, next) => {
+  const { userId, oldPassword, newPassword } = req.body;
+  getUserById(userId).then(({ rows }) => {
+    if (rows.length === 0) throw new Error();
+    return Promise.all([
+      compare(oldPassword, (rows[0].password)),
+      compare(newPassword, rows[0].password),
+    ]);
+  }).then(([oldPasswordMatches, newPasswordMatches]) => {
+    if (!oldPasswordMatches) throw new Error('You entered a wrong password');
+    else if (newPasswordMatches) throw new Error('Your new password and old password are the same');
+    hash(newPassword, 10);
```

Comment:
:open_mouth: Remember, the hash method returns a promise and changePassword query also returns a promise. What is the purpose of `hash()` if the promise is not returned?
Also, you are inserting the new password in the database without hashing it! :thinking: 
So, I suggest you return the hash promise so it will be resolved in the next then and then pass the hashed password to the query. This way:
```suggestion
    return hash(newPassword, 10);
    }).then(hashedPass => changePassword(userId, hahsedPass))
    .then(({rows})=> rows[0].password && res.send({ message: 'password is changed'}))
    .catch(erro => ....)
```

Reply:
Thank you I updated it ^_^